File: pointers/1_basic-pointers/2_address-vairables.c
#include <stdio.h>

int main(){

    int var = 10;

    printf("Base: %p\n",(void*)&var);

    printf("Base + 1: %p\n",(void*)(&var + 1));// 4

    printf("Base + 5: %p\n",(void*)(&var + 5)); //  20 bytes

    return 0;
}

File: pointers/1_basic-pointers/2_address-vairables.rs
fn main(){
    let var: i32 = 10;
    
    let base_ptr = &var as *const i32;

    unsafe{
        let next_locus = base_ptr.add(1);
         println!("Base: {:p}", base_ptr); //Base: 0x8a0fdfac4
         println!("Base + 1: {:p}", next_locus); //Base + 1: 0x8a0fdfac8

    }
}